### PR TITLE
Refact: restoreTeamsCmdF() - Error handling

### PR DESCRIPTION
**Pull request by dsharma522**
#### Ticket Link
Refactor - https://github.com/mattermost/mattermost-server/issues/21470
Jira- [MM-47816](https://mattermost.atlassian.net/browse/MM-47816)

### CODE REVIEW

File: commands/team.go [hunk 1]
```diff
@@ -354,16 +355,19 @@ func modifyTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 
 func restoreTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	teams := getTeamsFromTeamArgs(c, args)
+	var result *multierror.Error
 	for i, team := range teams {
 		if team == nil {
+			result = multierror.Append(result, errors.New("Unable to find team '"+args[i]+"'"))
```
Comment (isacikgoz) [inline commands/team.go:361]: We can use formatted errors for a better readability.
```suggestion
			result = multierror.Append(result, fmt.Errorf("unable to find team %s", args[i]))
```

**Review by isacikgoz — COMMENTED** [review]
This look good overall @dsharma522 just a small comment on error formatting.

File: commands/team.go (hunk 1 above)
Comment (dsharma522) [reply commands/team.go]: Thanks. I have accepted the suggestions.

File: commands/team.go [hunk 2]
```diff
@@ -354,16 +356,19 @@ func modifyTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
...
 	teams := getTeamsFromTeamArgs(c, args)
+	var result *multierror.Error
 	for i, team := range teams {
 		if team == nil {
+			result = multierror.Append(result, fmt.Errorf("unable to find team '%s'", args[i]))
 			printer.PrintError("Unable to find team '" + args[i] + "'")
 			continue
 		}
 		if rteam, _, err := c.RestoreTeam(team.Id); err != nil {
+			result = multierror.Append(result, errors.New("Unable to restore team '"+team.Name+"' error: "+err.Error()))
```
Comment (noxer) [inline commands/team.go:367]: Please use `fmt.Errorf` here as well, that way you can use string interpolation like in line 362.

File: commands/team_e2e_test.go [hunk 3]
```diff
@@ -451,9 +453,12 @@ func (s *MmctlE2ETestSuite) TestRestoreTeamsCmd() {
 		teamName := "non-existent-team"
 
 		err := restoreTeamsCmdF(c, &cobra.Command{}, []string{teamName})
-		s.Require().Nil(err)
+		var expected error
+		errMessage := "unable to find team '" + teamName + "'"
+		expected = multierror.Append(expected, errors.New(errMessage))
 
-		errMessage := "Unable to find team '" + teamName + "'"
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/team_e2e_test.go:460]: It's not necessary to check for the error separately, the next line will do this check as well.

File: commands/team_e2e_test.go [hunk 4]
```diff
@@ -466,9 +471,12 @@ func (s *MmctlE2ETestSuite) TestRestoreTeamsCmd() {
 		s.Require().Nil(appErr)
 
 		err := restoreTeamsCmdF(s.th.Client, &cobra.Command{}, []string{team.Name})
-		s.Require().Nil(err)
+		var expected error
+		errMessage := "unable to find team '" + team.Name + "'"
+		expected = multierror.Append(expected, errors.New(errMessage))
 
-		errMessage := "Unable to find team '" + team.Name + "'"
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/team_e2e_test.go:478]: Same here

File: commands/team_test.go [hunk 5]
```diff
@@ -848,8 +849,11 @@ func (s *MmctlUnitTestSuite) TestRestoreTeamsCmd() {
 			Times(1)
 
 		err := restoreTeamsCmdF(s.client, cmd, []string{"team1"})
-		s.Require().Nil(err)
-		s.Require().Equal("Unable to find team 'team1'", printer.GetErrorLines()[0])
+		var expected error
+		expected = multierror.Append(expected, fmt.Errorf("unable to find team '%s'", teamName))
+
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/team_test.go:855]: Same here

File: commands/team_test.go [hunk 6]
```diff
@@ -898,8 +902,10 @@ func (s *MmctlUnitTestSuite) TestRestoreTeamsCmd() {
 			Times(1)
 
 		err := restoreTeamsCmdF(s.client, cmd, []string{"team1"})
-		s.Require().Nil(err)
-		s.Require().Equal("Unable to restore team 'team1' error: an error occurred restoring a team",
-			printer.GetErrorLines()[0])
+		var expected error
+		expected = multierror.Append(expected, errors.New("Unable to restore team '"+teamName+"' error: an error occurred restoring a team"))
+
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/team_test.go:908]: Same here

**Review by noxer — CHANGES_REQUESTED** [review]
Thank you very much for your contribution @dsharma522, I'd like to request a few small changes, though.

File: commands/team.go (hunk 2 above)
Comment (dsharma522) [reply commands/team.go]: Done.

File: commands/team_e2e_test.go (hunk 3 above)
Comment (dsharma522) [reply commands/team_e2e_test.go]: Removed this check.

File: commands/team_e2e_test.go (hunk 4 above)
Comment (dsharma522) [reply commands/team_e2e_test.go]: Removed this check

File: commands/team_test.go (hunk 5 above)
Comment (dsharma522) [reply commands/team_test.go]: Removed check.

File: commands/team_test.go (hunk 6 above)
Comment (dsharma522) [reply commands/team_test.go]: Removed check and used string interpolations for error formatting.

**Review by noxer — CHANGES_REQUESTED** [review]
Great work, just one tiny change and it's ready from my side.

**Review by isacikgoz — APPROVED** [review]
Good to go after @noxer's comment addressed. Thanks for the contribution @dsharma522

File: commands/team.go [hunk 7]
```diff
@@ -364,7 +364,7 @@ func restoreTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error
 			continue
 		}
 		if rteam, _, err := c.RestoreTeam(team.Id); err != nil {
-			result = multierror.Append(result, errors.New("Unable to restore team '"+team.Name+"' error: "+err.Error()))
+			result = multierror.Append(result, fmt.Errorf("unable to restore team '%s' error: %s", team.Name, err.Error()))
```
[the remark this answers is not in the document]
Comment (dsharma522) [reply commands/team.go]: Thanks for suggestion. I have added change in this [commit](https://github.com/mattermost/mmctl/pull/581/commits/979e47625af2a9abec2b2555c569d3f4f39162ac)

**Review by noxer — APPROVED** [review]

**Pull request merged**